### DISCUSSION TITLE
Translations display should resolve nested translations

### DISCUSSION THREAD

**Discussion by mattiaz9** [directus/directus#9094] [Feature Requests]
The translations display works fine when the translations are related to the current item. Example:
`article --> translations`

but not for nested translations such as:
`article --> category --> translations`

Now, I tried to make my own display for this, which kinda works if it wasn't that I get this error: `Error: [useStores]: The stores could not be found.` after few items are displayed. That means that for the first 4-5 items I have no problem, but after that threshold I get the error, and that error also create problems when navigating other items.
Since I copied a few utilities not available in the sdks, I'm not sure if those utilities are the ones creating conflicts (eg: `getRelatedCollection`, `getDisplays`, `adjustFieldsForDisplays`, `getFieldsFromTemplate`, `notEmpty`, `useRelation`).

So what I'm really asking is if i's possibile to implement that feature in the original display.

Thanks.

**Comment by rijkvanzanten** [comment]
@mattiaz9 Would you like to open a PR to implement this functionality in the core display?

**Reply by mattiaz9** [reply]
So, I forked the repo, cloned and installed the `node_modules ` locally, but when I try to compile using `npm run dev` I get a lot of errors.
How do you compile (and also test) locally? is there any guide for this?

**Reply by rijkvanzanten** [reply]
Have you tried these steps? https://docs.directus.io/contributing/running-locally/

**Reply by mattiaz9** [reply]
@rijkvanzanten I made a pull request: https://github.com/directus/directus/pull/9776

Hope it's good for you!

**Reply by rijkvanzanten** [reply]
@mattiaz9 Thanks! Unfortunately, it was a little too overkill for what we're trying to achieve.. I think we should look into a way where we can rely on the current "one-display-per-level" approach that we're also relying on in the related-values display, rather than have a single top-level parent display recursively search through the whole tree of fields. 

When using `article --> category --> translations` for example, we should have the `related-values` display on `category`, and the `translations` display on `translations`, rather than trying to use the `translations` display directly on `category`, and have it recursively search through the rest of the tree to figure out what to render. Hope that makes sense!

**Reply by mattiaz9** [reply]
@rijkvanzanten ok, no problem! but could you at least fix this issue so I can build my custom translations display? https://github.com/directus/directus/discussions/10255

The problem with related values is that it doesn't resolve the translations and shows a list rather than the single value (I don't wanna click every time to see what category it is).

**Reply by rijkvanzanten** [reply]
> The problem with related values is that it doesn't resolve the translations and shows a list rather than the single value (I don't wanna click every time to see what category it is).

True! The ability to have the related values display show the first X items inline would be an nice improvement for the related values display 🙂 

@nickrum Shall we go ahead in providing the stores in those option functions, or is there a better way?

**Reply by nickrum** [reply]
Hmm, this is very specific to external/custom displays as it's not needed by the integrated ones. I'd rather keep the parameters of those functions as clean as possible.
Ideally we would call those option functions within the context of a Vue component (i.e. make them reactive composables). That way, `useStores()` could be used within those functions. This would likely introduce a breaking change though because the parameters would have to be made reactive.
Alternatively, we could allow for a function to be passed to `definedDisplay()` and pass the stores as a parameter of that function.

**Reply by mattiaz9** [reply]
@nickrum is it possible to directly export the stores from the extension sdk? would it work?

**Reply by nickrum** [reply]
Unfortunately, this is not easily possible because the stores are defined in the App package which is bundled and minified at build time.

**Reply by rijkvanzanten** [reply]
> deally we would call those option functions within the context of a Vue component (i.e. make them reactive composables)

I think this is already how it currently works!

**Reply by rijkvanzanten** [reply]
Aqui:

https://github.com/directus/directus/blob/cbe139d7e9602fb5abf7717f17544595b859f36c/app/src/modules/settings/routes/data-model/field-detail/shared/extension-options.vue#L79-L83

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
